### PR TITLE
User management: Add Viewers tab

### LINKED ISSUE #77814: Calypso: Viewer role users not searchable by email on simple sites
### Quick summary

The search feature in Calypso (WordPress.com > My Sites > Users > Search bar at the top) does not return accurate results when searching for viewers by email address or username on Simple Sites. This issue appears to be specific to the "Viewer" role, and it occurs regardless of the site's privacy setting. The problem does not replicate on Atomic sites.

### Steps to reproduce

prereq. Set the site to Private status, otherwise the Viewer role is not available.
*   Navigate to WordPress.com > My Sites > Users.
*   Ensure there are users assigned with the "Viewer" role.
*   Use the search bar at the top and attempt to find a Viewer by entering their email address or username.

### What you expected to happen

The search feature should return a list of users that match the entered email address or username, including users with the "Viewer" role.

### What actually happened

The search does not return any results when searching for Viewers by email address. It seems to be working inconsistently for usernames.

### Impact

Some (< 50%)

### Available workarounds?

[Yes, difficult to implement](https://github.com/Automattic/wp-calypso/issues/77814#issuecomment-1791996765)

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

The user who reported this issue: 6369117-zen

The issue appears to be specific to the "Viewer" role and is not affected by the site's privacy settings (private or public). It could be a bug within the Calypso user search functionality specifically related to Simple sites. More testing is needed to determine if this is an isolated issue or more widespread.

A video showing how I could reproduce this issue can be found here:

p1685986572045829-slack-C03TY6J1A

### PR DESCRIPTION
**Pull request by zaguiini**
Closes https://github.com/Automattic/wp-calypso/issues/77814.

## Proposed Changes

This PR separates viewers from the other types of users on a site. It adds a new tab called "Viewers", which is (will be, TODO item) searchable through e-mail or username.

image

## TO DO

- [x] Input placeholder
- [x] CTA color and text
- [x] Pre-select the "Viewer" role when clicking the CTA that sends the user to `/people/new/%s?role=viewer`
- [x] Check if the infinite query works
- [x] Remove the `include_viewers` query parameter from the `/sites/%s/users` endpoint (do we want to do that?)
- [ ] Update [support docs](https://wordpress.com/support/invite-people/#adding-and-removing-viewers-on-a-private-site) and heads-up HEs

## Testing Instructions

In a private site, navigate to `Users > All Users`.

Verify that the "Viewers" tab shows up.

Verify that a new user shows up in the list after adding a new user with the viewer role.

### CODE REVIEW

File: client/my-sites/people/people-list-section-header/index.jsx [hunk 1]
```diff
@@ -42,6 +43,10 @@ class PeopleListSectionHeader extends Component {
 			return '/people/email-followers/' + siteSlug;
 		}
 
+		if ( this.isViewersTab() ) {
+			return `/people/new/${ siteSlug }?role=viewer`;
```
Comment (zaguiini) [inline client/my-sites/people/people-list-section-header/index.jsx]: The idea is to pre-select the viewer role in the new user form. Let me know if there is already, or a better way, of doing that.

File: client/my-sites/people/people-list-section-header/index.jsx [hunk 2]
```diff
@@ -101,7 +112,7 @@ class PeopleListSectionHeader extends Component {
 								position="right"
 								text={ popoverText }
 								privacyLink={ false }
-								link="https://wordpress.com/support/followers/"
+								link={ localizeUrl( 'https://wordpress.com/support/followers/' ) }
```
Comment (zaguiini) [inline client/my-sites/people/people-list-section-header/index.jsx:123]: This is not related to this PR but it was a linter error that I fixed.

File: client/data/viewers/use-viewers-query.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			},
+		}
+	);
+
+	return {
+		data: {
+			...data,
+			viewers: uniqueBy( extractPages( data?.pages ?? [] ), compareUnique ),
+			total: data?.pages[ 0 ].found,
+		},
```
Comment (zaguiini) [inline client/data/viewers/use-viewers-query.ts:53]: It's not possible to modify the `data` type through `queryOptions.select` when using infinite queries.

I mean, it is possible to modify the runtime elements, but the type itself does not change: https://www.reddit.com/r/reactjs/comments/nh1s9n/comment/gz5e1ku/?utm_source=reddit&utm_medium=web2x&context=3

This is probably a bug in `react-query`, though. Anyways, this workaround fixes it.

**Review by zaguiini — COMMENTED** [review]
Do we want to leave both the "Remove" button and the chevron?

image

I feel it's appropriate to leave both, but the UI gets a little confusing. It's handy for the user to remove a viewer without entering their profile, and the chevron is a good hint that there's more to the list item. Not sure how to proceed in this case. @javierarce might chime in.

**Comment by vishnugopal** [comment]
Thanks much for working on this! 🤗 

> I feel it's appropriate to leave both, but the UI gets a little confusing

Let's remove it for consistency because the User tab also doesn't have it.

> clicking the CTA that sends the user to /people/new/%s?role=viewer

Think this is ok to do!


>  Remove the include_viewers query parameter from the /sites/%s/users endpoint (do we want to do that?)

Yes I feel we should do this! The implementation is buggy and it's not used anywhere else.

>  Check if the infinite query works

I checked with the user who reported the [bug here](6369117-zd-woothemes) and it works!

cc @bogiii for a review as well.

>  Search on the back-end

I feel it's ok to release a first version just disabling the search bar just to maintain parity with the current implementation.

We'll have to coordinate our changes with https://github.com/Automattic/wp-calypso/pull/78933

**Comment by javierarce** [comment]
> I feel it's appropriate to leave both, but the UI gets a little confusing. It's handy for the user to remove a viewer without entering their profile, and the chevron is a good hint that there's more to the list item. Not sure how to proceed in this case. @javierarce might chime in.

I agree with @vishnugopal, let's keep the remove button inside the profile like we do with subscribers. 

Also, at some point, the designs that @crisbusquets created for Subscribers (h8tMpJlCqNFemEerU9owHI-fi-812_37857) will be implemented, and users and viewers will need to inherit the same design language.

**Comment by crisbusquets** [comment]
> > I feel it's appropriate to leave both, but the UI gets a little confusing. It's handy for the user to remove a viewer without entering their profile, and the chevron is a good hint that there's more to the list item. Not sure how to proceed in this case. @javierarce might chime in.
> 
> I agree with @vishnugopal, let's keep the remove button inside the profile like we do with subscribers.
> 
> Also, at some point, the designs that @crisbusquets created for Subscribers ([h8tMpJlCqNFemEerU9owHI-fi-812_37857](https://href.li/?https://www.figma.com/file/h8tMpJlCqNFemEerU9owHI/?node-id=812-37857)) will be implemented, and users and viewers will need to inherit the same design language.

Thank you @javierarce !

Yep, the first milestone has been implemented, Apex team is building it.

**Comment by bogiii** [comment]
Hey @zaguiini,

I was doing the User Management Revamp, and all newly introduced changes were marked under the feature flag.
We have never removed the existing logic, and you should probably try to reuse it again.

You can see it providing query param: `?flags=-user-management-revamp`

Markup on 2023-07-06 at 12:11:44

Also, I remember that we introduced each card item to be clickable, which opens the details screen. The details screen will also contain the remove button so I would hide the remove button from there.

cc @vishnugopal

**Comment by zaguiini** [comment]
Thank you all for the reviews. I didn't have time to work on this today, but will get back to it ASAP.

**Comment by zaguiini** [comment]
Thanks, @bogiii, I found that my changes are actually reusing the introduced components, so it's not too different.

Alright, this should be good to go.

@vishnugopal I disabled the search for this component until we have it on the back end. I don't think I'll implement it now since there was never a way to search for viewers, and I'm not convinced it's worth the effort right now given my scarce time.

I created the PR to remove the `include_viewers` parameter: https://github.com/Automattic/jetpack/pull/31760. I'd appreciate it if you could shepherd this PR and the Jetpack PR into `trunk` and WPCOM.

Also, I'd like it if @Automattic/caribou could heads-up the HEs and modify the support docs as I will have different priorities next week. Thanks 🙏

**Comment by bogiii** [comment]
Hey @zaguiini,

Since we removed the subscriber's tab from the `All Users` section, are we forget to remove `Add Subscribers` as well?

Screenshot 2023-07-10 at 10 37 06

**Comment by zaguiini** [comment]
> Hey @zaguiini,
> 
> Since we removed the subscriber's tab from the `All Users` section, are we forget to remove `Add Subscribers` as well?
> 
> Screenshot 2023-07-10 at 10 37 06

Oh, maybe? I'm not sure if we want to do that. There is still a way to manage subscribers in Calypso through Users > Subscribers 🤔 Regardless, I do not have time to work on it, so I'd appreciate if you could shepherd this change and the PR too

**Comment by vishnugopal** [comment]
Closing as this is defunct now, we'll have to redo and rework this.

**Comment by michalczaplinski** [comment]
@zaguiini @vishnugopal @bogiii 

Could you elaborate on why this is defunct now? I'm trying to work on https://github.com/Automattic/wp-calypso/issues/77814, and this PR was supposed to be a temporary fix.

**Pull request closed without merging**